saw-central/src/SAWCentral/Options.hs

```diff
@@ -114,10 +114,10 @@ processEnv opts = do
       case mbJavaPath of
         Nothing       -> pure os
         Just javaPath -> do
-          javaMajorVersion <- findJavaMajorVersion javaPath
-          if javaMajorVersion >= 9
-             then pure os
-             else addRTJar javaPath os
+          mbJavaMajorVersion <- findJavaMajorVersion javaPath
+          case mbJavaMajorVersion of
+            Just v | v < 9 -> addRTJar javaPath os
+            _ -> pure os
```

Comment:
I think we should emit a warning here if we find Java on the `PATH` but `findJavaMajorVersion` returns `Nothing`, since that indicates that something is misconfigured. (This is different from the advice I suggested on the `crucible-jvm` side in https://github.com/GaloisInc/crucible/pull/1457#discussion_r2191081103, but that's because this code path is reachable even if you aren't doing any JVM verification.)